### PR TITLE
Adjust Subscription pages to support TierType instead of Group

### PR DESCRIPTION
**Pull request by emilwidlund**
Refactors the Subscription UI to support the new TierType approach instead of the old SubscriptionGroup API.

### CODE REVIEW

**Comment by emilwidlund** [comment]
@frankie567 Mind taking a look at this refactor? It seems to be working; but I just want a second opinion on how I approached this here :)

File: clients/apps/web/src/components/Subscriptions/SubscriptionGroupIcon.tsx [hunk 1]
```diff
@@ -17,26 +20,28 @@ const SubscriptionGroupIcon: React.FC<SubscriptionGroupIconProps> = ({
    * so we could set any icon dynamically.
    */
```
Comment (frankie567) [inline clients/apps/web/src/components/Subscriptions/SubscriptionGroupIcon.tsx]: This comment doesn't make sense anymore since we have a fixed enum :)

**Comment by frankie567** [comment]
LGTM, great job 🙏

File: clients/apps/web/src/components/Subscriptions/SubscriptionGroupIcon.tsx (hunk 1 above)
Comment (emilwidlund) [reply clients/apps/web/src/components/Subscriptions/SubscriptionGroupIcon.tsx]: Aye! Good catch!

**Pull request merged**